### ISSUE TITLE
Timeout fatal error during cookies retrieval

### ISSUE THREAD

**Issue by HiLivin** [AlexCSDev/PatreonDownloader#40]
It's my first time using the app. I've logged in the downloaded browser and now the app exits and I'm getting this error:

```
C:\Users\user\Downloads\net3.1-win-x64-release>PatreonDownloader.App.exe --url "https://www.patreon.com/m/xxx/posts" --download-directory c:\xxx --descriptions --embeds --campaign-images --json --verbose
2021-01-16 16:15:37.9460 DEBUG Initializing PuppeteerEngine with parameters , True
2021-01-16 16:15:38.0084 DEBUG Found 15 chrome processes (not sure which one yet)
2021-01-16 16:15:38.0240 INFO Retrieving cookies...
2021-01-16 16:15:38.0396 DEBUG Calling login check
2021-01-16 16:15:38.0552 DEBUG Retrieving browser
2021-01-16 16:15:38.0552 DEBUG Downloading browser
2021-01-16 16:15:38.0708 DEBUG Launching browser
2021-01-16 16:15:38.8330 DEBUG Opening new page
2021-01-16 16:15:39.1542 DEBUG Creating IWebBrowser
2021-01-16 16:15:39.2834 DEBUG Checking login status
2021-01-16 16:15:44.1108 DEBUG We are logged in
2021-01-16 16:15:44.1368 DEBUG Retrieving browser
2021-01-16 16:15:44.1368 DEBUG Retrieving cookies
2021-01-16 16:16:17.5231 FATAL Fatal error, application will be closed: PuppeteerSharp.NavigationException: Timeout of 30000 ms exceeded
 ---> System.TimeoutException: Timeout of 30000 ms exceeded
   at PuppeteerSharp.Helpers.TaskHelper.<>c__DisplayClass2_0.<WithTimeout>b__0()
   at PuppeteerSharp.Helpers.TaskHelper.WithTimeout(Task task, Func`1 timeoutAction, TimeSpan timeout, CancellationToken cancellationToken)
   at PuppeteerSharp.FrameManager.NavigateFrameAsync(Frame frame, String url, NavigationOptions options)
   --- End of inner exception stack trace ---
   at PuppeteerSharp.FrameManager.NavigateFrameAsync(Frame frame, String url, NavigationOptions options)
   at PatreonDownloader.PuppeteerEngine.Wrappers.Browser.WebPage.GoToAsync(String url, Nullable`1 timeout, WaitUntilNavigation[] waitUntil) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.PuppeteerEngine\Wrappers\Browser\WebPage.cs:line 30
   at PatreonDownloader.PuppeteerEngine.PuppeteerCookieRetriever.RetrieveCookies() in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.PuppeteerEngine\PuppeteerCookieRetriever.cs:line 124
   at PatreonDownloader.App.Program.RunPatreonDownloader(String url, Boolean headlessBrowser, Uri remoteBrowserAddress, PatreonDownloaderSettings settings) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.App\Program.cs:line 135
   at PatreonDownloader.App.Program.Main(String[] args) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.App\Program.cs:line 74
```

**Comment by AlexCSDev** [comment]
Try running app with --no-headless parameter. You should your patreon home page inside browser window when the app tells you "DEBUG Retrieving cookies".

* If you see this page does the loading spinner on the page's tab go away or spins until the app gives you timeout error?
* If you see another page what page is it?

**Comment by HiLivin** [comment]
Alright,  after adding "--no-headless" it started to suck the content.
After downloading about a hundred posts it gave me this kind of error:

```
2021-01-17 10:41:13.4155 FATAL Fatal error, application will be closed: System.NullReferenceException: Object reference not set to an instance of an object.
   at PatreonDownloader.Engine.DefaultPlugin.ExtractSupportedUrls(String htmlContents) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.Engine\DefaultPlugin.cs:line 192
   at PatreonDownloader.Engine.PluginManager.ExtractSupportedUrls(String htmlContents) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.Engine\PluginManager.cs:line 122
   at PatreonDownloader.Engine.Stages.Crawling.PageCrawler.ParsePage(String json, Boolean saveDescriptions, Boolean saveEmbeds, String downloadDirectory) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.Engine\Stages\Crawling\PageCrawler.cs:line 181
   at PatreonDownloader.Engine.Stages.Crawling.PageCrawler.Crawl(CampaignInfo campaignInfo, PatreonDownloaderSettings settings, String downloadDirectory) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.Engine\Stages\Crawling\PageCrawler.cs:line 77
   at PatreonDownloader.Engine.PatreonDownloader.Download(String url, PatreonDownloaderSettings settings) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.Engine\PatreonDownloader.cs:line 240
   at PatreonDownloader.App.Program.RunPatreonDownloader(String url, Boolean headlessBrowser, Uri remoteBrowserAddress, PatreonDownloaderSettings settings) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.App\Program.cs:line 161
   at PatreonDownloader.App.Program.Main(String[] args) in E:\Sources\BigProjects\PatreonDownloader\PatreonDownloader.App\Program.cs:line 74
```

**Comment by AlexCSDev** [comment]
>Alright, after adding "--no-headless" it started to suck the content.
That means this was a temporary problem, you don't need to use --no-headless anymore.

>After downloading about a hundred posts it gave me this kind of error:
I will need your json files (produced by --json parameter) to figure out what is going on. You can send them to alexcsdev@protonmail.com

**Comment by AlexCSDev** [comment]
Fixed in 0.9.4.1
